aws/resource_aws_lakeformation_datalake_settings_test.go

```diff
@@ -0,0 +1,116 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lakeformation"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAWSLakeFormationDataLakeSettings_basic(t *testing.T) {
+	callerIdentityName := "data.aws_caller_identity.current"
+	resourceName := "aws_lakeformation_datalake_settings.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSLakeFormationDataLakeSettingsEmpty,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLakeFormationDataLakeSettingsConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourceAttrAccountID(resourceName, "catalog_id"),
+					resource.TestCheckResourceAttrPair(callerIdentityName, "account_id", resourceName, "catalog_id"),
+					resource.TestCheckResourceAttr(resourceName, "admins.#", "1"),
+					testAccCheckAWSLakeFormationDataLakePrincipal(callerIdentityName, "arn", resourceName, "admins"),
+				),
+			},
+		},
+	})
+}
+
+const testAccAWSLakeFormationDataLakeSettingsConfig_basic = `
+data "aws_caller_identity" "current" {}
+
+resource "aws_lakeformation_datalake_settings" "test" {
+  admins = ["${data.aws_caller_identity.current.arn}"]
+}
+`
+
+func TestAccAWSLakeFormationDataLakeSettings_withCatalogId(t *testing.T) {
+	callerIdentityName := "data.aws_caller_identity.current"
+	resourceName := "aws_lakeformation_datalake_settings.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSLakeFormationDataLakeSettingsEmpty,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLakeFormationDataLakeSettingsConfig_withCatalogId,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckResourceAttrAccountID(resourceName, "catalog_id"),
+					resource.TestCheckResourceAttrPair(callerIdentityName, "account_id", resourceName, "catalog_id"),
```

Comment:
Can use `testAccCheckResourceAttrAccountID()` here.

Reply:
You mean checks at l. 57 and l. 58 are redundant? Same as above, I can remove the last one, IMO there is no need to check `aws_caller_identity` resource alone.